### PR TITLE
[NEW] SFU Server

### PR DESCRIPTION
**Pull request by hrahul2605**
## Proposed changes (including videos or screenshots)

New SFU Server for handling voice channels

### CODE REVIEW

**Comment by hrahul2605** [comment]
@dougfabris

**Comment by hrahul2605** [comment]
Closing this, continued [here](https://github.com/hrahul2605/Rocket.Chat/pull/2)

File: package.json [hunk 1]
```diff
@@ -253,6 +257,7 @@
 		"postis": "^2.2.0",
 		"prom-client": "^12.0.0",
 		"prometheus-gc-stats": "^0.6.2",
+		"protoo-server": "^4.0.6",
```
Comment (dougfabris) [inline package.json:260]: This package is a kind of WebSocket? Do you think it's possible to use the package we already have?

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (hrahul2605) [reply package.json:260]: I'm not sure which package you are referring here, `websockets` or `ws` both of them are not available here

Comment (dougfabris) [reply package.json:260]: I think is a build-in package in meteor, but let's skip that for now

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: sfu/config.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		{
+			mediaCodecs:
+			[
+				{
+					kind: 'audio',
+					mimeType: 'audio/opus',
+					clockRate: 48000,
+					channels: 2,
+				},
+				{
```
[the remark this answers is not in the document]
Comment (hrahul2605) [reply sfu/config.ts:59]: nope, these were defaults, I'll update this

**Review by dougfabris — APPROVED** [review]

**Pull request merged**